### PR TITLE
tweaked hunger and thirst descriptions

### LINKED ISSUE #29430: New satiety mechanic needs fixing
**Describe the bug**  
A character with no hunger or thirst status, i.e. neutral.
I give him a couple of deluxe beans & rice - nothing happens.
I give him clean water like 6 or 7 times - suddenly he is sated.
I hit the wait button. The 'sated' status keeps flicking on and off every other turn.
I give him some more solid food - the 'sated' status disappears.

Is this just working on a delay, or what's going on?

**Versions and configuration(please complete the following information):**
 - OS: Win 8.1 x64
 - Game Version: 8746
 - Graphics version: Tiles

### PR DESCRIPTION
**Pull request by KorGgenT**
#### Summary

```SUMMARY: Interface "Tweak hunger (and thirst) descriptions in HUD to be more useful"```
#### Purpose of change

lots of complaints recently about the hunger logic. I decided to take another stab at making the hunger description in the sidebar be more useful for telling you when you really need to eat, at least in part.  thirst is easy enough, just reduce the delay. in fact, it has been brought to my attention that it's easier to not drink enough in one sitting than the opposite. as far as hunger is concerned, i added a new word "peckish" to denote that you could eat, but probably don't need to. "very hungry" is reserved to being a relatively long time since you ate with a calorie surplus, or a relatively short time if you have a calorie deficit. "famished" "starving" and "near starving" only appear if you have a calorie deficit.
Fixes #29430

### CODE REVIEW

**Comment by kevingranade** [comment]
See Jenkins:
```
01:37:03 CCACHE_CPP2=1 ccache clang++  -DRELEASE -DGIT_VERSION -DBACKTRACE -DLOCALIZE -ffast-math -Os  -Wall -Wextra    -fsigned-char -Werror -std=c++11 -MMD -MP -m32 -D_GNU_SOURCE -I/usr/include/ncursesw -c src/activity_handlers.cpp -o obj/activity_handlers.o
01:37:04 In file included from src/action.cpp:17:
01:37:04 In file included from src/output.h:11:
01:37:04 src/player.h:858:42: error: 'get_hunger_description' overrides a member function but is not marked 'override' [-Werror,-Winconsistent-missing-override]
01:37:04         std::pair<std::string, nc_color> get_hunger_description() const;
01:37:04                                          ^
01:37:04 src/character.h:227:50: note: overridden virtual function is here
01:37:04         virtual std::pair<std::string, nc_color> get_hunger_description() const;
01:37:04                                                  ^
01:37:04 1 error generated.
01:37:04 Makefile:768: recipe for target 'obj/action.o' failed
01:37:04 make: *** [obj/action.o] Error 1
01:37:04 make: *** Waiting for unfinished jobs....
01:37:09 In file included from src/activity_handlers.cpp:28:
01:37:09 In file included from src/output.h:11:
01:37:09 src/player.h:858:42: error: 'get_hunger_description' overrides a member function but is not marked 'override' [-Werror,-Winconsistent-missing-override]
01:37:09         std::pair<std::string, nc_color> get_hunger_description() const;
01:37:09                                          ^
01:37:09 src/character.h:227:50: note: overridden virtual function is here
01:37:09         virtual std::pair<std::string, nc_color> get_hunger_description() const;
01:37:09                                                  ^
01:37:09 1 error generated.
01:37:09 Makefile:768: recipe for target 'obj/activity_handlers.o' failed
01:37:09 make: *** [obj/activity_handlers.o] Error 1
```

**Comment by KorGgenT** [comment]
meant to open this as a draft. marking it WIP until i've had a good chance to playtest (and fix that jenkins error)

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/getting-very-hungery-with-over-60-000-calories/19704/8

**Comment by KorGgenT** [comment]
jenkins rebuild

**Comment by KorGgenT** [comment]
https://github.com/KorGgenT/Cataclysm-DDA/releases/tag/hunger
for distributable

File: src/player.cpp [hunk 1]
```diff
@@ -12703,3 +12703,63 @@ void player::do_skill_rust()
         }
     }
 }
+
+std::pair<std::string, nc_color> player::get_hunger_description() const
```
Comment (kevingranade) [inline src/player.cpp]: I think this needed to be player::hunger_stat()

Comment (KorGgenT) [reply src/player.cpp]: no but thanks for point at this, i wouldn't have noticed that hunger_stat was a duplicate until after i was doing more with the UI

**Comment by esotericist** [comment]
So, a couple of days of light play with this, and my primary feedback is: the stuff you currently use to indicate 'calorie deficit' makes more sense to me if it only gets used when you are on a downward trend. i.e. your average calories per day over the last few days is lower than your average calories over, say, the last month or so.

Rationale: human experience is qualitative/comparative, and will tend to be within the context of "how I have been" not "how I should be".

I don't know how feasible this is, nor am I especially wedded to timeframes, but I think feedback would be more meaningful if it functioned in that comparative fashion. Because if someone is at some consistent place well below their ideal weight for a while, that becomes 'normal' for them.

**Comment by KorGgenT** [comment]
i think it's reasonable to say that is a pretty good place for this to sit, as that's pretty well outside of the scope of this PR, and even outside of the scope of my larger project with related things. we may revisit sometime later to get new normals,

**Comment by esotericist** [comment]
> i think it's reasonable to say that is a pretty good place for this to sit, as that's pretty well outside of the scope of this PR, and even outside of the scope of my larger project with related things. we may revisit sometime later to get new normals,

Fair enough. The new descriptions are definitely an improvement and make it *possible* for someone to know "I am eating enough" once they're at their ideal range for calories, and we didn't have that prior to this PR

**Pull request merged**